### ISSUE TITLE
SPMC layer problem

### ISSUE THREAD

**Issue by karenyun** [jiangsutx/SPMC_VideoSR#14]
@jiangsutx Hi, I have a question about the SPMC layer, why you use the Flow(i->0) as backward warping to warp the I(0) which aligned with the I(i) instead of the I(0), since the final output of the whole network is the I(H0)?

Could you take some time to give some explanations? Thanks very much!

**Comment by jiangsutx** [comment]
We calculate Flow(i->0) and use forward warping to warp i-frame to 0-frame.
So that all frames are aligned to 0-frame.

The reason is according to imaging model. Please refer to the paper for more details.

**Comment by karenyun** [comment]
@jiangsutx , thanks for quick reply!
In the paper the "Training Strategy" "Phase 1": you use I(L, 0-i) aligned with I(L, i) to compute the loss, so I firstly consider that the SPMC warps the 0-frame to i-frame. 

Another question is that in the "Figure 3 (b)", it seems that -T ~ T LR frames all contribute the original HR(i)  image, instead of just using warped I(L, i) image to produce the HR(i)?  I am just confused about this part, could you give some explanations again? Thanks!

**Comment by jiangsutx** [comment]
Yes. We use all the neighboring frames to recover one center frame.
